app/routes/user_settings.py

```diff
@@ -63,30 +62,25 @@ def set_name():
 
 @server.route("/preferences/avatar", methods=['POST'])
 def set_avatar():
-    if g.user is None:
-        return abort(403)
     try:
         new_avatar_source = request.form['avatar']
     except KeyError:
         return abort(400)
-    try:
-        new_avatar = imgur_upload(new_avatar_source)
-    except RuntimeError:
-        return abort(400)
-    g.user.avatar = new_avatar
-    db.session.commit()
-    return do_redirect()
+
+    return user_settings.set_avatar(new_avatar_source) or do_redirect()
 
 
 @server.route("/preferences/profile", methods=['POST'])
 def set_profile_preferences():
     try:
         new_email = request.form['email']
         new_name = request.form['name']
+        avatar_url = request.form['avatar-url']
```

Comment:
Instead of this, what about doing `request.form.get(..., None)`, and have the controllers handle the `None` case (no change). That way, only the things being changed need to be POSTed.

Reply:
Since it's an HTML form however, it will submit the fields regardless so I don't have control over which fields are submitted

Reply:
Fair enough